modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipServiceImpl.java

```diff
@@ -83,6 +84,43 @@ public ObjectRelationship deleteObjectRelationship(
 			objectRelationshipPersistence.findByPrimaryKey(
 				objectRelationshipId);
 
+		ObjectDefinition lAccountsObjectDefinition =
+			_objectDefinitionPersistence.findByERC_C(
+				"L_ACCOUNT", objectRelationship.getCompanyId());
+
+		if (Objects.equals(
+				objectRelationship.getObjectDefinitionId1(),
+				lAccountsObjectDefinition.getObjectDefinitionId())) {
+
+			ObjectDefinition lPostalAddressObjectDefinition =
+				_objectDefinitionPersistence.findByERC_C(
+					"L_POSTAL_ADDRESS", objectRelationship.getCompanyId());
+
+			List<ObjectRelationship> allPostalAddressObjectRelationships =
+				objectRelationshipLocalService.getAllObjectRelationships(
+					lPostalAddressObjectDefinition.getObjectDefinitionId());
+
+			for (ObjectRelationship postalAddressObjectRelationship :
+					allPostalAddressObjectRelationships) {
+
+				if (Objects.equals(
+						postalAddressObjectRelationship.
+							getParameterObjectFieldId(),
+						objectRelationship.getObjectFieldId2())) {
+
+					_objectDefinitionModelResourcePermission.check(
+						getPermissionChecker(),
+						postalAddressObjectRelationship.
+							getObjectDefinitionId1(),
+						ActionKeys.UPDATE);
+
+					objectRelationshipLocalService.deleteObjectRelationship(
+						postalAddressObjectRelationship.
+							getObjectRelationshipId());
+				}
+			}
+		}
+
```

Comment:
The `ServiceImpl` is used to validate only permissions, all business rules needs to be implement in `LocalServiceImpl`.  In this case, we need to move this code to ObjectRelationshipLocalServiceImpl.

I believe that we can simplify this code.
1. Create a finder:
```xml
<finder name="ParameterObjectFieldId" return-type="Collection">
	<finder-column name="parameterObjectFieldId" />
</finder>
```
2. Create a logic to delete all related object relationships, something like this:
```java
private void _deleteObjectRelationshipsByParameterFieldId(
	long parameterFieldId) throws PortalException {

	for (ObjectRelationship objectRelationship :
		objectRelationshipPersistence.fetchByParameterObjectFieldId(
			parameterFieldId)) {

		objectRelationshipLocalService.deleteObjectRelationship(
			objectRelationship);
	}
}
```
Call this private method inside this [conditional](https://github.com/liferay/liferay-portal/blob/15d73820441b17d44f71b6c4a1faedbc44297a5a/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectRelationshipLocalServiceImpl.java#L322-L331), something like this:

```java
_deleteObjectRelationshipsByParameterFieldId(
				objectRelationship.getObjectFieldId2());
```